iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamOAuth2RequestFactory.java

```diff
@@ -164,4 +186,17 @@ public TokenRequest createTokenRequest(Map<String, String> requestParameters,
     return new TokenRequest(requestParameters, clientId, scopes, grantType);
   }
 
+  private boolean isValidUrl(String url) {
+    try {
+      new URL(url).toURI();
```

Comment:
Does this check that no fragments are present in the url? and what about a query part?

Reply:
No, just added the check on query parameters + fragments, and tests.